db/migrate/20200810194943_change_selectable_avatars_site_setting.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class ChangeSelectableAvatarsSiteSetting < ActiveRecord::Migration[6.0]
+  def up
+    selectable_avatars = execute("SELECT value FROM site_settings WHERE name = 'selectable_avatars'")
+    return if selectable_avatars.cmd_tuples == 0
+
+    # Keep old site setting value as a backup
+    execute <<~SQL
+      UPDATE site_settings
+      SET name = 'selectable_avatars_urls'
+      WHERE name = 'selectable_avatars'
+    SQL
+
+    # Extract SHA1s from URLs and then use them for upload ID lookups.
+    urls = []
+    sha1s = []
+    selectable_avatars.first["value"].split("\n").each do |url|
+      match = url.match(/(\/original\/\dX[\/\.\w]*\/(\h+)[\.\w]*)/)
+      if match.present?
+        urls << match[1]
+        sha1s << match[2]
+      else
+        STDERR.puts "Could not extract a SHA1 from #{url}"
+      end
+    end
+
+    upload_ids = execute <<~SQL
+      SELECT id
+      FROM uploads
+      WHERE url IN (#{urls.map { |url| "'#{url}'" }.join(',')})
```

Comment:
What happens if either `urls` or `sha1s` is empty?

Reply:
It generated an invalid query `SELECT id FROM uploads WHERE url IN () OR sha1 IN ()`, but I fixed it.

```
discourse_development=# SELECT id FROM uploads WHERE url IN () OR sha1 IN ();
ERROR:  syntax error at or near ")"
LINE 1: SELECT id FROM uploads WHERE url IN () OR sha1 IN ();
                                             ^
```